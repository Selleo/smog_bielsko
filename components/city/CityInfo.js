import React, { Component, PropTypes }       from 'react';
import { View, Image, StyleSheet, Animated } from 'react-native';

import CityDescription                       from './CityDescription'
import CityName                              from './CityName'
import header                                from '../stylesheets/Header'
import { fetchBgr }                          from './../sharing/Extends'

export default class CityInfo extends Component {
  constructor(props) {
    super(props);
    this.state = {
      fetched: false,
      fadeAnim: new Animated.Value(0),
    }
  }

  componentDidMount() {
    let queryName;

    if (this.props.stationName == undefined) queryName = this.getCityInfo(0) + ' town';
    else queryName = this.props.stationName + ' town';

    fetchBgr(queryName)
      .then((response) => {
        this.setState({ bgr: response.link });
        Animated.timing(
          this.state.fadeAnim,
          {
            toValue: 1,
            duration: 700
          }
        ).start();
      })
      .catch((response) => {
        console.log(response);
      })
  }

  secondUmageStyles() {
    return {
      alignSelf: 'center',
      backgroundColor: 'transparent',
      borderWidth: 0,
      flex: 1,
      flexDirection: 'column',
      justifyContent: 'center',
      margin: 0,
      opacity: this.state.fadeAnim,
      padding: 0,
      resizeMode: 'cover',
      ...StyleSheet.absoluteFillObject
    }
  }

  render() {
    let { bgr } = this.state;
    return (
      <View style={header.container}>
        <Image source={require('./../images/header.jpg')} style={header.bgr}>
          <Animated.Image source={{ uri: bgr }} style={this.secondUmageStyles()}/>
          <View style={{ alignItems: 'flex-start'}}>
            <Image source={require('./../images/header_overlay.png')} style={[header.bgr, header.bgr2]}/>
            <CityName text={this.getCityInfo(0)}/>
            <CityDescription text={this.getCityInfo(1)}/>
          </View>
        </Image>
      </View>
    )
  }

  getCityInfo(part) {
    return this.props.city.name.split(', ')[part]
  }

  static propTypes = {
    city: PropTypes.object.isRequired,
  };
}
