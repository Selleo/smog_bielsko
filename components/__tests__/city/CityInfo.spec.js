import 'react-native';
import React                          from 'react';

import { shallow }                    from 'enzyme';
import chai, { expect }               from 'chai';
import dirtyChai                      from 'dirty-chai';

import CityInfo                       from '../../city/CityInfo';

chai.use(dirtyChai);

let wrapper, props;

describe('CityInfo', () => {
  it('renders correctly', () => {
    props = { city: { name: 'Zywiec, ul. Rynek 2' } };
    wrapper = shallow(<CityInfo {...props}/>);

    expect(wrapper).to.exist();
  });

  it('renders description on html', () => {
    props = { city: { name: 'Zywiec, ul. Rynek 2' } };
    wrapper = shallow(<CityInfo {...props}/>);

    expect(wrapper.children().length).to.equal(2);
    expect(wrapper.children().nodes[0].type.name).to.equal('CityName');
    expect(wrapper.children().nodes[1].type.name).to.equal('CityDescription');
  });

});
