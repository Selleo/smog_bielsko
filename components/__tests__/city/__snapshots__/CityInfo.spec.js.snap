exports[`CityInfo renders correctly 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <View
    style={
        Object {
            "height": 180,
            "position": "relative",
            "width": undefined,
          }
    }>
    <Image
        source={1}
        style={
            Object {
                "alignSelf": "stretch",
                "backgroundColor": "transparent",
                "borderWidth": 0,
                "flex": 1,
                "flexDirection": "column",
                "height": 180,
                "justifyContent": "flex-end",
                "margin": 0,
                "padding": 0,
                "paddingBottom": 10,
                "top": 0,
                "width": undefined,
              }
        }>
        <AnimatedComponent
            source={
                Object {
                    "uri": undefined,
                  }
            }
            style={
                Object {
                    "alignSelf": "center",
                    "backgroundColor": "transparent",
                    "borderWidth": 0,
                    "bottom": 0,
                    "flex": 1,
                    "flexDirection": "column",
                    "justifyContent": "center",
                    "left": 0,
                    "margin": 0,
                    "opacity": 0,
                    "padding": 0,
                    "position": "absolute",
                    "resizeMode": "cover",
                    "right": 0,
                    "top": 0,
                  }
            } />
        <View
            style={
                Object {
                    "alignItems": "flex-start",
                  }
            }>
            <Image
                source={1}
                style={
                    Array [
                        Object {
                          "alignSelf": "stretch",
                          "backgroundColor": "transparent",
                          "borderWidth": 0,
                          "flex": 1,
                          "flexDirection": "column",
                          "height": 180,
                          "justifyContent": "flex-end",
                          "margin": 0,
                          "padding": 0,
                          "paddingBottom": 10,
                          "top": 0,
                          "width": undefined,
                        },
                        Object {
                          "bottom": 0,
                          "height": 180,
                          "left": 0,
                          "position": "absolute",
                          "resizeMode": "stretch",
                          "right": 0,
                          "top": -16,
                        },
                      ]
                } />
            <CityName
                text="Bielsko-Biała" />
            <CityDescription
                text="ul. Kossak-Szczuckiej 19" />
        </View>
    </Image>
</View>,
  "nodes": Array [
    <View
      style={
            Object {
                  "height": 180,
                  "position": "relative",
                  "width": undefined,
                }
      }>
      <Image
            source={1}
            style={
                  Object {
                        "alignSelf": "stretch",
                        "backgroundColor": "transparent",
                        "borderWidth": 0,
                        "flex": 1,
                        "flexDirection": "column",
                        "height": 180,
                        "justifyContent": "flex-end",
                        "margin": 0,
                        "padding": 0,
                        "paddingBottom": 10,
                        "top": 0,
                        "width": undefined,
                      }
            }>
            <AnimatedComponent
                  source={
                        Object {
                              "uri": undefined,
                            }
                  }
                  style={
                        Object {
                              "alignSelf": "center",
                              "backgroundColor": "transparent",
                              "borderWidth": 0,
                              "bottom": 0,
                              "flex": 1,
                              "flexDirection": "column",
                              "justifyContent": "center",
                              "left": 0,
                              "margin": 0,
                              "opacity": 0,
                              "padding": 0,
                              "position": "absolute",
                              "resizeMode": "cover",
                              "right": 0,
                              "top": 0,
                            }
                  } />
            <View
                  style={
                        Object {
                              "alignItems": "flex-start",
                            }
                  }>
                  <Image
                        source={1}
                        style={
                              Array [
                                    Object {
                                      "alignSelf": "stretch",
                                      "backgroundColor": "transparent",
                                      "borderWidth": 0,
                                      "flex": 1,
                                      "flexDirection": "column",
                                      "height": 180,
                                      "justifyContent": "flex-end",
                                      "margin": 0,
                                      "padding": 0,
                                      "paddingBottom": 10,
                                      "top": 0,
                                      "width": undefined,
                                    },
                                    Object {
                                      "bottom": 0,
                                      "height": 180,
                                      "left": 0,
                                      "position": "absolute",
                                      "resizeMode": "stretch",
                                      "right": 0,
                                      "top": -16,
                                    },
                                  ]
                        } />
                  <CityName
                        text="Bielsko-Biała" />
                  <CityDescription
                        text="ul. Kossak-Szczuckiej 19" />
            </View>
      </Image>
</View>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <CityInfo
        city={
                Object {
                        "geo": Array [
                          49.813464,
                          19.027318,
                        ],
                        "name": "Bielsko-Biała, ul. Kossak-Szczuckiej 19",
                        "url": "http://aqicn.org/city/poland/slaski/bielsko-biala--ul.-kossak-szczuckiej-19/",
                      }
        } />,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": CityInfo {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "city": Object {
            "geo": Array [
              49.813464,
              19.027318,
            ],
            "name": "Bielsko-Biała, ul. Kossak-Szczuckiej 19",
            "url": "http://aqicn.org/city/poland/slaski/bielsko-biala--ul.-kossak-szczuckiej-19/",
          },
        },
        "refs": Object {},
        "state": Object {
          "fadeAnim": 0,
          "fetched": false,
        },
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <View
          style={
                    Object {
                              "height": 180,
                              "position": "relative",
                              "width": undefined,
                            }
          }>
          <Image
                    source={1}
                    style={
                              Object {
                                        "alignSelf": "stretch",
                                        "backgroundColor": "transparent",
                                        "borderWidth": 0,
                                        "flex": 1,
                                        "flexDirection": "column",
                                        "height": 180,
                                        "justifyContent": "flex-end",
                                        "margin": 0,
                                        "padding": 0,
                                        "paddingBottom": 10,
                                        "top": 0,
                                        "width": undefined,
                                      }
                    }>
                    <AnimatedComponent
                              source={
                                        Object {
                                                  "uri": undefined,
                                                }
                              }
                              style={
                                        Object {
                                                  "alignSelf": "center",
                                                  "backgroundColor": "transparent",
                                                  "borderWidth": 0,
                                                  "bottom": 0,
                                                  "flex": 1,
                                                  "flexDirection": "column",
                                                  "justifyContent": "center",
                                                  "left": 0,
                                                  "margin": 0,
                                                  "opacity": 0,
                                                  "padding": 0,
                                                  "position": "absolute",
                                                  "resizeMode": "cover",
                                                  "right": 0,
                                                  "top": 0,
                                                }
                              } />
                    <View
                              style={
                                        Object {
                                                  "alignItems": "flex-start",
                                                }
                              }>
                              <Image
                                        source={1}
                                        style={
                                                  Array [
                                                            Object {
                                                              "alignSelf": "stretch",
                                                              "backgroundColor": "transparent",
                                                              "borderWidth": 0,
                                                              "flex": 1,
                                                              "flexDirection": "column",
                                                              "height": 180,
                                                              "justifyContent": "flex-end",
                                                              "margin": 0,
                                                              "padding": 0,
                                                              "paddingBottom": 10,
                                                              "top": 0,
                                                              "width": undefined,
                                                            },
                                                            Object {
                                                              "bottom": 0,
                                                              "height": 180,
                                                              "left": 0,
                                                              "position": "absolute",
                                                              "resizeMode": "stretch",
                                                              "right": 0,
                                                              "top": -16,
                                                            },
                                                          ]
                                        } />
                              <CityName
                                        text="Bielsko-Biała" />
                              <CityDescription
                                        text="ul. Kossak-Szczuckiej 19" />
                    </View>
          </Image>
</View>,
        "_debugID": 2,
        "_renderedOutput": <View
          style={
                    Object {
                              "height": 180,
                              "position": "relative",
                              "width": undefined,
                            }
          }>
          <Image
                    source={1}
                    style={
                              Object {
                                        "alignSelf": "stretch",
                                        "backgroundColor": "transparent",
                                        "borderWidth": 0,
                                        "flex": 1,
                                        "flexDirection": "column",
                                        "height": 180,
                                        "justifyContent": "flex-end",
                                        "margin": 0,
                                        "padding": 0,
                                        "paddingBottom": 10,
                                        "top": 0,
                                        "width": undefined,
                                      }
                    }>
                    <AnimatedComponent
                              source={
                                        Object {
                                                  "uri": undefined,
                                                }
                              }
                              style={
                                        Object {
                                                  "alignSelf": "center",
                                                  "backgroundColor": "transparent",
                                                  "borderWidth": 0,
                                                  "bottom": 0,
                                                  "flex": 1,
                                                  "flexDirection": "column",
                                                  "justifyContent": "center",
                                                  "left": 0,
                                                  "margin": 0,
                                                  "opacity": 0,
                                                  "padding": 0,
                                                  "position": "absolute",
                                                  "resizeMode": "cover",
                                                  "right": 0,
                                                  "top": 0,
                                                }
                              } />
                    <View
                              style={
                                        Object {
                                                  "alignItems": "flex-start",
                                                }
                              }>
                              <Image
                                        source={1}
                                        style={
                                                  Array [
                                                            Object {
                                                              "alignSelf": "stretch",
                                                              "backgroundColor": "transparent",
                                                              "borderWidth": 0,
                                                              "flex": 1,
                                                              "flexDirection": "column",
                                                              "height": 180,
                                                              "justifyContent": "flex-end",
                                                              "margin": 0,
                                                              "padding": 0,
                                                              "paddingBottom": 10,
                                                              "top": 0,
                                                              "width": undefined,
                                                            },
                                                            Object {
                                                              "bottom": 0,
                                                              "height": 180,
                                                              "left": 0,
                                                              "position": "absolute",
                                                              "resizeMode": "stretch",
                                                              "right": 0,
                                                              "top": -16,
                                                            },
                                                          ]
                                        } />
                              <CityName
                                        text="Bielsko-Biała" />
                              <CityDescription
                                        text="ul. Kossak-Szczuckiej 19" />
                    </View>
          </Image>
</View>,
      },
      "_renderedNodeType": 1,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <CityInfo
    city={
        Object {
            "geo": Array [
              49.813464,
              19.027318,
            ],
            "name": "Bielsko-Biała, ul. Kossak-Szczuckiej 19",
            "url": "http://aqicn.org/city/poland/slaski/bielsko-biala--ul.-kossak-szczuckiej-19/",
          }
    } />,
}
`;
